### PR TITLE
[RW-5113][risk=no] UI - Sidebar to right of page with functionality to add criteria to groups (phase 1)

### PR DESCRIPTION
**Pull request by NehaBroad**
As part of this PR:

1. Criteria icon will replace the workspace icon in sidebar only if the user is on the cohort criteria search page

2. By Default the the count on the icon will be 0. 

3. Selecting the plus icon will update the count and show a message 'Criteria Added' for few seconds

Screen Shot 2020-07-20 at 12 56 25 PM

4. Clicking the icon will open the add modifier page (the look and feel for this will be taken care by another Story/PR)
Screen Shot 2020-07-15 at 5 02 05 PM

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/components/help-sidebar.tsx [hunk 1]
```diff
@@ -363,6 +381,18 @@ export const HelpSidebar = fp.flow(withCurrentWorkspace(), withUserProfile())(
       }
     }
 
+    renderCriteria() {
+      const {criteria} = this.props;
+      return <PopupTrigger
```
Comment (dolbeew) [inline ui/src/app/components/help-sidebar.tsx:386]: I think the selections are meant to slide out, like how the help tips slide out when the info icon is clicked. We may want to verify before changing, but I believe that's what was discussed.

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx [hunk 2]
```diff
@@ -144,9 +145,10 @@ interface Props {
   selections: Array<Selection>;
   setView: Function;
   view: string;
+  criteria?: Array<Selection>;
 }
 
-export class SelectionList extends React.Component<Props> {
+export const SelectionList = withCurrentCohortCriteria()(class extends React.Component<Props> {
```
Comment (dolbeew) [inline ui/src/app/cohort-search/selection-list/selection-list.component.tsx:151]: The changes in this component will break the non feature flagged version of cohort builder. We either need to put the changes behind the feature flag or duplicate the component so we can change it however we want. Duplicating is probably simpler.

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx [hunk 3]
```diff
@@ -5,16 +5,19 @@ import {AttributesPage} from 'app/cohort-search/attributes-page/attributes-page.
...
 import {FlexRowWrap} from 'app/components/flex';
 import {ClrIcon} from 'app/components/icons';
 import {SpinnerOverlay} from 'app/components/spinners';
-import colors, {addOpacity} from 'app/styles/colors';
+import colors, {addOpacity, colorWithWhiteness} from 'app/styles/colors';
 import {reactStyles, ReactWrapperBase} from 'app/utils';
 import {triggerEvent} from 'app/utils/analytics';
+import {currentCohortCriteriaStore} from 'app/utils/navigation';
 import {Criteria, CriteriaType, DomainType, TemporalMention, TemporalTime} from 'generated/fetch';
+import {Messages} from 'primereact/messages';
```
Comment (dolbeew) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:20]: Just curious, did you also look at Primereact's [Growl component](https://primefaces.org/primereact/showcase/#/growl)? At a glance, it looks more similar to what the mocks call for, not sure if it's actually a better fit for what we need.

File: ui/src/app/components/help-sidebar.tsx [hunk 4]
```diff
@@ -177,7 +194,7 @@ const icons = (helpContentKey: string) => {
     faIcon: helpContentKey === NOTEBOOK_HELP_CONTENT ? faFolderOpen : faInfoCircle,
     label: helpContentKey === NOTEBOOK_HELP_CONTENT ? 'Storage Icon' : 'Help Icon',
     page: null,
-    style: {fontSize: '21px'},
+    style: {fontSize: '1.1rem'},
```
Comment (dolbeew) [inline ui/src/app/components/help-sidebar.tsx:197]: This style is only applied to the info icon

File: ui/src/app/pages/data/cohort-review/cohort-review.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ interface State {
   readonly: boolean;
 }
 
-export class CohortReview extends React.Component<{}, State> {
+export const CohortReview = withCurrentCohortCriteria()(class extends React.Component<{}, State> {
```
[the remark this answers is not in the document]
Comment (NehaBroad) [reply ui/src/app/pages/data/cohort-review/cohort-review.tsx:29]: this is just to use withCurrentCohortCriteria

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx (hunk 2 above)
Comment (NehaBroad) [reply ui/src/app/cohort-search/selection-list/selection-list.component.tsx:151]: yeah thats a good point , i will update this

File: ui/src/app/components/help-sidebar.tsx (hunk 1 above)
Comment (NehaBroad) [reply ui/src/app/components/help-sidebar.tsx:386]: Ok, wasn't aware of this. I'll change it

Comment (dolbeew) [reply ui/src/app/components/help-sidebar.tsx:386]: Ok, let me know if you need to discuss, but should look similar to [this block](https://github.com/all-of-us/workbench/pull/3788/files#diff-52586f24a61602acc0422d94891b3666R530-R532) in the template and then add a new object for the icon to [icons array](https://github.com/all-of-us/workbench/pull/3788/files#diff-52586f24a61602acc0422d94891b3666L173). The hiding and showing of the content should already be built in.

File: ui/src/app/pages/data/cohort-review/cohort-review.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (dolbeew) [reply ui/src/app/pages/data/cohort-review/cohort-review.tsx:29]: Ok, was asking because I don't think we'll need the current criteria here. Cohort review only uses cohorts that have been saved, so we shouldn't need access to any unsaved selections.

File: ui/src/app/components/help-sidebar.tsx (hunk 1 above)
Comment (NehaBroad) [reply ui/src/app/components/help-sidebar.tsx:386]: As discussed offline , i'll take care of this in another PR

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx (hunk 3 above)
Comment (NehaBroad) [reply ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:20]: Growl might be a better option for this as itdisplay message in overlay vs messages that are supposed to be for inline. I'll change to Growl

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx [hunk 6]
```diff
@@ -164,26 +167,34 @@ export class SelectionList extends React.Component<Props> {
...
+      {!serverConfigStore.getValue().enableCohortBuilderV2 && <div style={styles.buttonContainer}>
         <Button type='link'
           style={{...styles.button, color: colors.dark, fontSize: '14px'}}
           onClick={() => close()}>
           Cancel
         </Button>
         {this.showNext && <Button type='primary' onClick={() => setView('modifiers')}
           style={styles.button}
-          disabled={selections.length === 0}>
+          disabled={!criteria || criteria.length === 0}>
```
Comment (dolbeew) [inline ui/src/app/cohort-search/selection-list/selection-list.component.tsx:197]: Need to revert this change also, the Finish button stays disabled after selections are made in the non feature-flagged UI

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx [hunk 7]
```diff
@@ -193,10 +204,10 @@ export class SelectionList extends React.Component<Props> {
         </Button>}
         <Button type='primary' onClick={() => finish()}
           style={styles.button}
-          disabled={selections.length === 0 || disableFinish}>
+          disabled={!criteria || criteria.length === 0 || disableFinish}>
```
Comment (dolbeew) [inline ui/src/app/cohort-search/selection-list/selection-list.component.tsx:207]: Same as above, need to revert this

File: ui/src/app/components/help-sidebar.tsx (hunk 4 above)
Comment (dolbeew) [reply ui/src/app/components/help-sidebar.tsx:197]: If you use inspect mode on the Invision screens, looks like both should be ~21px font size

File: ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx [hunk 8]
```diff
@@ -104,7 +104,51 @@ interface State {
   treeSearchTerms: string;
 }
 
+const css = `
```
Comment (dolbeew) [inline ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:107]: May need to tweak/add some styles here to match the mock, can do in the follow-up PR if you prefer. Looks like mainly the check icon is too big and increases the the total height of the growl

Comment (NehaBroad) [reply ui/src/app/cohort-search/cohort-search/cohort-search.component.tsx:107]: I'll take care of this in follow-up PR for criteria pop up
